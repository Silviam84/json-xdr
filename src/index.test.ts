import { toJSON } from './index'
import * as XDR from 'js-xdr';

const types = XDR.config();

XDR.config((xdr) => {
  xdr.struct('Color', [
    ['red', xdr.int()],
    ['green', xdr.int()],
    ['blue', xdr.int()]
  ]);
}, types);

describe('#toJSON', function() {
  test('converts XDR to JSON', () => {
    let color = new types.Color({
      red: 0,
      green: 1,
      blue: 2
    });

    expect(toJSON(types, types.Color, color)).toMatchSnapshot()
  });
})
