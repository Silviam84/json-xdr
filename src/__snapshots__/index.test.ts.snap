// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toJSON converts XDR to JSON 1`] = `
Object {
  "blue": 2,
  "green": 1,
  "red": 0,
}
`;
